lib/parse-cron/all/parse-cron.rbi

```diff
@@ -0,0 +1,134 @@
+# typed: strict
+
+# Parses cron expressions and computes the next occurence of the "job"
+#
+class CronParser
+  extend T::Sig
+  extend T::Generic
+
+  # NOTE: Make these variant if you use the alternate sigs for later methods.
+  SourceClass = type_member(fixed: T.class_of(Time))
+  Source = type_member(fixed: Time)
```

Comment:
It looks like you're trying to define a type alias. Why are generics required here?

```suggestion
  SourceClass = T.type_alias {T.class_of(Time)}
  Source = T.type_alias {Time}
```

also it's not clear to me that these type aliases are pulling their weight. Does the parse-cron job actually use Sorbet and define these type_members / type_aliases? If not, by adding them, Sorbet will think that this is ok:

```
puts CronParser::Source
```

when in that's a NameError because that constant doesn't exist at runtime.

`Time` is fewer characters than `Source`. I'd just inline `Time` into all the sigs, and same with `T.class_of(Time)`

Reply:
Thanks for reviewing this!

parse-cron doesn't use Sorbet. These generics/aliases are from me trying to fully type this behavior:
```ruby
ExtendedTime = Class.new(Time)
result = CronParser.new("* * * * *", ExtendedTime).next
result.class == ExtendedTime
```
I don't use that at all, though. I was mostly going for completeness (mostly to play with Sorbet).

I didn't realize that Sorbet would think the names given to type members are valid constants. I thought they'd be invisible during type checking outside of that class.

Given all this, I agree that it makes sense to inline them. I'll do that and include a small test file to show that this at least covers the basic cases.